<a href="https://colab.research.google.com/github/Jaejuna/SpeedWagon/blob/main/dataextract/Data_extract(5_5%EB%A7%8C).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd

In [3]:
train = pd.read_csv('/content/drive/MyDrive/BOAZ_miniProject2/train.csv')
valid = pd.read_csv('/content/drive/MyDrive/BOAZ_miniProject2/valid.csv')

In [4]:
print("Size of train data :", len(train))
print("Size of validation data :", len(valid))

Size of train data : 279992
Size of validation data : 35004


In [5]:
train['Topic'].unique()

array(['주거와생활', '시사교육', '상거래(쇼핑)', '개인및관계',
       '행사', '미용과건강', '일과직업', '식음료', '여가생활'],
      dtype=object)

In [6]:
# Topic 별 Dialogue 수
pd.DataFrame(train.groupby("Topic").count().iloc[:,0])

Text
Topic                
개인및관계   71130
미용과건강   17069
상거래(쇼핑)  26298
시사교육       13119
식음료        30184
여가생활      35247
일과직업     20428
주거와생활    45179
행사           21338

In [7]:
pd.DataFrame(valid.groupby("Topic").count().iloc[:,0])

Text
Topic               
개인및관계   8892
미용과건강   2134
상거래(쇼핑)  3288
시사교육       1640
식음료        3773
여가생활      4407
일과직업     2554
주거와생활    5648
행사           2668

# Data extract
- 토픽 9개
- train : 279992 -> 55986
  - 개인및관계(14226), 미용과건강(3412), 상거래(5258), 시사교육(2622), 식음료(6036), 여가생활(7048), 일과직업(4084), 주거와생활(9034), 행사(4266)
- valid : 35004 -> 6992
  - 개인및관계(1778), 미용과건강(426), 상거래(656), 시사교육(328), 식음료(754), 
  여가생활(880), 일과직업(510), 주거와생활(1128), 행사(532)  
- 랜덤으로 추출
 

## train data

In [8]:
df1 = train.loc[train['Topic']=='주거와생활'].sample(n=9034)
df2 = train.loc[train['Topic']=='시사교육'].sample(n=2622)
df3 = train.loc[train['Topic']=='상거래(쇼핑)'].sample(n=5258)
df4 = train.loc[train['Topic']=='개인및관계'].sample(n=14226)
df5 = train.loc[train['Topic']=='행사'].sample(n=4266)
df6 = train.loc[train['Topic']=='미용과건강'].sample(n=3412)
df7= train.loc[train['Topic']=='일과직업'].sample(n=4084)
df8 = train.loc[train['Topic']=='식음료'].sample(n=6036)
df9 = train.loc[train['Topic']=='여가생활'].sample(n=7048)

In [9]:
 train_new = pd.concat([df1,df2,df3,df4,df5,df6,df7,df8,df9], ignore_index=True)

In [14]:
len(train_new)

55986

## valid data

In [12]:
df1 = valid.loc[valid['Topic']=='주거와생활'].sample(n=1128)
df2 = valid.loc[valid['Topic']=='시사교육'].sample(n=328)
df3 = valid.loc[valid['Topic']=='상거래(쇼핑)'].sample(n=656)
df4 = valid.loc[valid['Topic']=='개인및관계'].sample(n=1778)
df5 = valid.loc[valid['Topic']=='행사'].sample(n=532)
df6 = valid.loc[valid['Topic']=='미용과건강'].sample(n=426)
df7= valid.loc[valid['Topic']=='일과직업'].sample(n=510)
df8 = valid.loc[valid['Topic']=='식음료'].sample(n=754)
df9 = valid.loc[valid['Topic']=='여가생활'].sample(n=880)

In [13]:
valid_new = pd.concat([df1,df2,df3,df4,df5,df6,df7,df8,df9], ignore_index=True)

In [15]:
len(valid_new)

6992

## save

In [16]:
train_new.to_csv('/content/drive/MyDrive/BOAZ_miniProject2/train_new_2.csv', index = False)

In [17]:
valid_new.to_csv('/content/drive/MyDrive/BOAZ_miniProject2/valid_new_2.csv', index = False)